src/native/external/zlib/dotnet_allocator_unix.c

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
We want `external\zlib` directory to have minimal delta against upstream. Ideally, these files should not be under zlib directory since you are not planning to upstream them.

If you want to keep these files here, they should be mentioned in https://github.com/dotnet/runtime/blob/main/src/native/external/zlib-version.txt.